api/user_test.go

```diff
@@ -2339,3 +2339,81 @@ func TestAutocompleteUsers(t *testing.T) {
 		t.Fatal("should have errored - bad team id")
 	}
 }
+
+func TestGetByUsername(t *testing.T) {
+	th := Setup().InitBasic()
+	Client := th.CreateClient()
+
+	user := model.User{Email: strings.ToLower(model.NewId()) + "yomamma@simulator.amazonses.com", Nickname: "Jabba The Hutt", Password: "passwd1"}
+	ruser, _ := Client.CreateUser(&user, "")
```

Comment:
If you use the BasicInit there's no need for this, etc.. please take a look at what the BasicInit does